Managers/PrimitiveMan.h

```diff
@@ -0,0 +1,546 @@
+			/// Translates coordinates from scene to this bitmap offset producing two coordinates.
+			/// 'Left' scene bitmap with negative values as if scene seam is 0,0 and 'Right' scene bitmap with positive values.
+			/// </summary>
+			/// <param name="targetPos">Target position.</param>
+			/// <param name="scenePos">Position in scene.</param>
+			/// <param name="drawLeftPos"></param>
+			/// <param name="drawRightPos"></param>
+			/// <remarks>
+			/// Unfortunately it's hard to explain how this works. It tries to represent scene bitmap as two parts with center in 0,0.
+			/// Right part is just plain visible part with coordinates from [0, scenewidth] and left part is imaginary bitmap as if we traversed it across the seam right-to-left with coordinates [0, -scenewidth].
+			/// So in order to be drawn each screen coordinates calculated twice for left and right 'bitmaps' and then one of them either flies away off-screen or gets drawn on the screen.
+			/// When we cross the seam either left or right part is actually drawn in the bitmap, and negative coordinates of right part are compensated by view point offset coordinates when we cross the seam right to left.
+			/// I really don't know how to make it simpler, because it has so many special cases and simply wrapping all out-of-the scene coordinates don't work because this way nothing will be ever draw across the seam.
+			/// You're welcome to rewrite this nightmare if you can, I wasted a whole week on this (I can admit that I'm just too dumb for this) )))
+			/// </remarks>
+			void TranslateCoordinates(Vector targetPos, Vector scenePos, Vector & drawLeftPos, Vector & drawRightPos);
+
+			/// <summary>
+			/// Draws this primitive on provided bitmap.
+			/// </summary>
+			/// <param name="pDrawScreen">Bitmap to draw on.</param>
+			/// <param name="targetPos">Position of graphical primitive.</param>
+			virtual void Draw(BITMAP *pDrawScreen, Vector targetPos) = 0;
+		};
+#pragma endregion
+
+#pragma region Line Primitive
+		/// <summary>
+		/// Class used to schedule drawing of line primitives created from Lua.
+		/// </summary>
+		class LinePrimitive : public GraphicalPrimitive {
+
+		public:
+
+			/// <summary>
+			/// Constructor method for LinePrimitive object.
+			/// </summary>
+			LinePrimitive(Vector start, Vector end, unsigned short color) {
+				m_StartPos = start;
+				m_EndPos = end;
+				m_Color = color;
+				m_Player = -1;
+			}
+
+			/// <summary>
+			/// Constructor method for LinePrimitive object.
+			/// </summary>
+			LinePrimitive(short player, Vector start, Vector end, unsigned short color) {
+				m_StartPos = start;
+				m_EndPos = end;
+				m_Color = color;
+				m_Player = player;
+			}
+
+			/// <summary>
+			/// Draws this primitive on provided bitmap.
+			/// </summary>
+			/// <param name="pDrawScreen">Bitmap to draw on.</param>
+			/// <param name="targetPos">Position of graphical primitive.</param>
+			virtual void Draw(BITMAP *pDrawScreen, Vector targetPos);
```

Comment:
I'm pretty sure I do because `GraphicalPrimitive::Draw` is a pure abstract method and I need virtual overrides for it to work.
This is how this was set up and I don't want to rewrite it.

Reply:
Oh yeah, wasn't saying to not make it pure abstract, it's correct to do it this way. Was hoping you could not define the Draw methods here and just implement them in the cpp file, but I tested it and it doesn't work, which makes sense cause the cpp file isn't part of the object definition.